server/src/main/java/keywhiz/service/daos/AclDAO.java

```diff
@@ -135,103 +153,156 @@ public void findAndEvictClient(long clientId, long groupId) {
+  }
+
   public ImmutableSet<SanitizedSecret> getSanitizedSecretsFor(Client client) {
     checkNotNull(client);
-    ImmutableSet.Builder<SanitizedSecret> sanitizedSet = ImmutableSet.builder();
-    SecretContentDAO secretContentDAO = createSecretContentDAO();
-
-    for (SecretSeries series : getSecretSeriesFor(client)) {
-      for (SecretContent content : secretContentDAO.getSecretContentsBySecretId(series.getId())) {
-        SecretSeriesAndContent seriesAndContent = SecretSeriesAndContent.of(series, content);
-        sanitizedSet.add(SanitizedSecret.fromSecretSeriesAndContent(seriesAndContent));
+    return dslContext.transactionResult(configuration -> {
+      ImmutableSet.Builder<SanitizedSecret> sanitizedSet = ImmutableSet.builder();
+
+      for (SecretSeries series : getSecretSeriesFor(client)) {
+        for (SecretContent content : secretContentDAO.getSecretContentsBySecretId(series.getId())) {
+          SecretSeriesAndContent seriesAndContent = SecretSeriesAndContent.of(series, content);
+          sanitizedSet.add(SanitizedSecret.fromSecretSeriesAndContent(seriesAndContent));
+        }
       }
-    }
-    return sanitizedSet.build();
+      return sanitizedSet.build();
+    });
   }
 
-  @SqlQuery("SELECT clients.id, clients.name, clients.description, clients.createdAt, clients.createdBy, " +
-            "clients.updatedAt, clients.updatedBy, clients.enabled, clients.automationAllowed FROM clients " +
-            "JOIN memberships ON clients.id = memberships.clientId " +
-            "JOIN accessGrants ON memberships.groupId = accessGrants.groupId " +
-            "JOIN secrets ON secrets.id = accessGrants.secretId " +
-            "WHERE secrets.name = :name")
-  public abstract Set<Client> getClientsFor(@BindSecret Secret secret);
+  public Set<Client> getClientsFor(Secret secret) {
+    List<Client> r = dslContext
+        .select(CLIENTS.ID, CLIENTS.NAME, CLIENTS.DESCRIPTION, CLIENTS.CREATEDAT, CLIENTS.CREATEDBY,
+            CLIENTS.UPDATEDAT, CLIENTS.UPDATEDBY, CLIENTS.ENABLED, CLIENTS.AUTOMATIONALLOWED)
+        .from(CLIENTS)
+        .join(MEMBERSHIPS).on(CLIENTS.ID.eq(MEMBERSHIPS.CLIENTID))
+        .join(ACCESSGRANTS).on(MEMBERSHIPS.GROUPID.eq(ACCESSGRANTS.GROUPID))
+        .join(SECRETS).on(SECRETS.ID.eq(ACCESSGRANTS.SECRETID))
+        .where(SECRETS.NAME.eq(secret.getName()))
+        .fetch()
+        .map(new ClientJooqMapper());
+    return new HashSet<>(r);
+  }
 
-  @Transaction
   public Optional<SanitizedSecret> getSanitizedSecretFor(Client client, String name, String version) {
     checkNotNull(client);
     checkArgument(!name.isEmpty());
     checkNotNull(version);
 
-    Optional<SecretSeries> secretSeries = getSecretSeriesFor(client, name);
-    if (!secretSeries.isPresent()) {
-      return Optional.empty();
-    }
+    // Cast to fix issue with mvn + java8 (not showing up in Intellij).
+    return (Optional<SanitizedSecret>)dslContext.transactionResult(configuration -> {
```

Comment:
I was planning to isolate the issues in a small piece of code and get people to comment on things.